cdap-common/src/main/java/co/cask/cdap/common/logging/LocationAwareWrapperLogger.java

```diff
@@ -25,16 +26,32 @@
 /**
  * A {@link Logger} implementation that only emit logs that are accepted by {@link LogSampler}.
  */
-final class LocationAwareSamplingLogger implements Logger, LocationAwareLogger {
+final class LocationAwareWrapperLogger implements Logger, LocationAwareLogger {
 
-  private static final String FQCN = LocationAwareSamplingLogger.class.getName();
+  private static final String FQCN = LocationAwareWrapperLogger.class.getName();
 
   private final Logger logger;
   private final LogSampler sampler;
+  private final String mdcKey;
+  private final String mdcValue;
 
-  LocationAwareSamplingLogger(Logger logger, LogSampler sampler) {
```

Comment:
Don't we need the sampling logger any more?

Would it be better to have an abstract class, like DelegatingLocationAwareLogger, which has the log() methods and implements all the methods. But defines before() and after() as abstract.

Then you can have a SamplingLogger that implements before() using a LogSampler, and a TaggingLogger that sets and unsets an MDC field in before() and after()